include/aws/io/private/pkcs11_private.h

```diff
@@ -15,5 +17,47 @@
 
 AWS_EXTERN_C_BEGIN
 
+/**
+ * Find the slot that meets all criteria:
+ * - has a token
+ * - if match_slot_id is non-null, then slot IDs must match
+ * - if match_token_label is non-null, then labels must match
+ * The function fails unless it finds exactly one slot meeting all criteria.
+ */
+AWS_IO_API
+int aws_pkcs11_lib_find_slot_with_token(
+    struct aws_pkcs11_lib *pkcs11_lib,
+    const unsigned long *match_slot_id,
+    const struct aws_string *match_token_label,
```

Comment:
they don't need to be null-terminated

it was just more convenient to pass `aws_string*` than `aws_byte_cursor*`, because everywhere that calls this function is storing an `aws_string*`, and this is a private function

if this was made into a public function, it should use `aws_byte_cursor*` instead.